utils/extra.py

```diff
@@ -93,28 +95,28 @@ async def post(bot, code):
         "embed_colour": "#FFFFFF",
     }
 
-    response = await bot.session.post(
+    async with await bot.session.post(
         "https://api.senarc.org/paste",
         json=paste_body,
         headers={"accept": "application/json", "Content-Type": "application/json"},
-    )
-    response = await response.json()
-    return response.get("url")
+    ) as response:
+        data = await response.json()
+        return data.get("url")
 
 
-async def get_paste(bot, paste_id):
-    response = await bot.session.get(
+async def get_paste(bot, paste_id) -> typing.Optional[str]:
+    async with bot.session.get(
         f"https://api.senarc.org/bin/{paste_id}", headers={"accept": "application/json", "headless": "true"}
-    )
-    response = await response.json()
-    return response.get("content")
+    ) as response:
+        data = await response.json()
+        return data.get("content")
 
 
-def random_history(data, number):
+def random_history(data, number) -> list[typing.Any]:
```

Comment:
wrong typing should be
`(data: Sequence[T], number: int) -> list[T]:`

or `list` instead of `Sequence` if its always a list

but TBH overall redundant function so just dont have it in the first place

Reply:
it's not cause I grab the entire random_history from the database on startup
then I sample it afterwards with certain amounts

Reply:
Fixed

Reply:
Also handled (except for removing)